sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -67,12 +67,7 @@ case class ShuffledHashJoinExec(
 
   // Exposed for testing
   @transient lazy val ignoreDuplicatedKey = joinType match {
-    case LeftExistence(_) =>
-      // For building hash relation, ignore duplicated rows with same join keys if:
-      // 1. Join condition is empty, or
-      // 2. Join condition only references streamed attributes and build join keys.
-      val streamedOutputAndBuildKeys = AttributeSet(streamedOutput ++ buildKeys)
-      condition.forall(_.references.subsetOf(streamedOutputAndBuildKeys))
+    case LeftExistence(_) if condition.isEmpty => true
```

Comment:
Thanks for fixing this issue!

Instead of disable `ignoreDuplicatedKey` in this case, Is it possible to relax the requirement to that the references in conditions are all in `streamedOutput ++ buildKeysThatAreAttributes` ? E.g. diff: 
```diff
-      val streamedOutputAndBuildKeys = AttributeSet(streamedOutput ++ buildKeys)
+      val attrBuildKeys = buildKeys.filter(_.isInstanceOf[Attribute])
+      val streamedOutputAndBuildKeys = AttributeSet(streamedOutput ++ attrBuildKeys)”
```
The current master branch allows `buildKeys`’ references in the `condition` (`AttributeSet` extracts references), while this diff limits it to `streamedOutput` and `buildKeys` that are `Attribute`s

Reply:
Or maybe something like:
```
      val streamedOutputSet = AttributeSet(streamedOutput)
      val buildKeysSet = ExpressionSet(buildKeys)
      condition.forall { c =>
        var valid = true
        c.transformDownWithPruning(c => valid && !buildKeysSet.contains(c.asInstanceOf[Expression])) {
          case a: Attribute if !streamedOutputSet.contains(a) =>
            valid = false
            a
        }
        valid
      }
```
to allow full expression of `buildKeys` to appear in condition.

Reply:
Yes, exactly. We can write a recursive function if that's easier to grasp.